packages/graph/hash_graph/lib/graph/src/types/schema/property_type.rs

```diff
@@ -0,0 +1,727 @@
+                    },
+                },
+                "required": ["https://example.com/property_type"]
+            })
+        );
+
+        let object2 = serde_json::from_value(json)?;
+        assert_eq!(object, object2);
+
+        Ok(())
+    }
+
+    #[test]
+    fn validate_object() {
+        let json = json!({
+            "type": "object",
+            "properties": {
+                "https://example.com/property_type": {
+                    "$ref": "https://example.com/property_type"
+                },
+            },
+            "required": ["https://example.com/property_type"]
+        });
+        assert!(serde_json::from_value::<PropertyTypeObject>(json).is_ok());
+
+        let json = json!({
+            "type": "object",
+            "properties": {
+                "https://example.com/property_type": {
+                    "$ref": "https://example.com/property_type"
+                },
+            },
+            "required": ["https://example.com/property_type_2"]
+        });
+        assert!(serde_json::from_value::<PropertyTypeObject>(json).is_err());
+    }
+
+    #[test]
+    fn property_value_object() -> Result<(), Box<dyn Error>> {
+        let object = PropertyValues::PropertyTypeObject(PropertyTypeObject {
+            properties: [(
+                Uri::new("https://example.com/property_type")?,
+                ValueOrArray::Value(PropertyTypeReference::new(Uri::new(
+                    "https://example.com/property_type",
+                )?)?),
+            )]
+            .into_iter()
+            .collect(),
+            required: vec![],
+        });
+
+        let json = serde_json::to_value(&object)?;
+        assert_eq!(
+            json,
+            json!({
+                "type": "object",
+                "properties": {
+                    "https://example.com/property_type": {
+                        "$ref": "https://example.com/property_type"
+                    },
+                }
+            })
+        );
+
+        let object2 = serde_json::from_value(json)?;
+        assert_eq!(object, object2);
+
+        Ok(())
+    }
+
+    #[test]
+    fn property_value_reference() -> Result<(), Box<dyn Error>> {
+        let object = PropertyValues::DataTypeReference(DataTypeReference::new(Uri::new(
+            "https://example.com/property_type",
+        )?)?);
+
+        let json = serde_json::to_value(&object)?;
+        assert_eq!(
+            json,
+            json!({
+                "$ref": "https://example.com/property_type"
+            })
+        );
+
+        let object2 = serde_json::from_value(json)?;
+        assert_eq!(object, object2);
+
+        Ok(())
+    }
+
+    #[test]
+    fn property_values() -> Result<(), Box<dyn Error>> {
+        let object = PropertyValues::ArrayOfPropertyValues(Array::new(
+            OneOf::new([PropertyValues::DataTypeReference(DataTypeReference::new(
+                Uri::new("https://example.com/property_type")?,
```

Comment:
As above, this URI is `example.com/property_type` on a `DataTypeReference`

Reply:
Applied in b32cbd7f62203be45bcbb9e755f1b03d6f297112